.editorconfig

```
@@ -0,0 +1,447 @@
+csharp_style_prefer_range_operator = true:suggestion
+csharp_style_implicit_object_creation_when_type_is_apparent = false:suggestion
+csharp_style_prefer_tuple_swap = true:suggestion
+csharp_style_prefer_utf8_string_literals = true:suggestion
+csharp_style_inlined_variable_declaration = true:suggestion
+csharp_style_deconstructed_variable_declaration = true:suggestion
+csharp_style_unused_value_assignment_preference = discard_variable:hint
+csharp_style_unused_value_expression_statement_preference = discard_variable:hint
+csharp_style_allow_blank_lines_between_consecutive_braces_experimental = false:error
+csharp_style_allow_embedded_statements_on_same_line_experimental = false:error
+csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental = true:error
+csharp_style_allow_blank_line_after_token_in_conditional_expression_experimental = true:silent
+csharp_style_allow_blank_line_after_token_in_arrow_expression_clause_experimental = true:silent
+csharp_preferred_modifier_order = public, private, protected, internal, static, extern, new, virtual, abstract, sealed, override, readonly, unsafe, volatile, async:error
+csharp_style_var_when_type_is_apparent = false:error
+csharp_style_prefer_switch_expression = true:suggestion
+csharp_indent_case_contents_when_block = false
+csharp_indent_case_contents = false
+csharp_indent_switch_labels = true
+csharp_style_var_for_built_in_types = false:error
+csharp_style_var_elsewhere = false:error
+csharp_style_prefer_extended_property_pattern = false:warning
+
+[*.{cs,vb}]
+#### Naming styles ####
+
+# Naming rules
+
+dotnet_naming_rule.interface_should_be_begins_with_i.severity = error
+dotnet_naming_rule.interface_should_be_begins_with_i.symbols = interface
+dotnet_naming_rule.interface_should_be_begins_with_i.style = begins_with_i
+
+dotnet_naming_rule.types_should_be_pascal_case.severity = error
+dotnet_naming_rule.types_should_be_pascal_case.symbols = types
+dotnet_naming_rule.types_should_be_pascal_case.style = pascal_case
+
+dotnet_naming_rule.non_field_members_should_be_pascal_case.severity = error
+dotnet_naming_rule.non_field_members_should_be_pascal_case.symbols = non_field_members
+dotnet_naming_rule.non_field_members_should_be_pascal_case.style = pascal_case
+
+dotnet_naming_rule.const_field_members_should_be_pascal_case.severity = error
+dotnet_naming_rule.const_field_members_should_be_pascal_case.symbols = const_field_members
+dotnet_naming_rule.const_field_members_should_be_pascal_case.style = pascal_case
+
+dotnet_naming_rule.local_var_and_method_args.severity = error
+dotnet_naming_rule.local_var_and_method_args.symbols = local_var_and_method_args
+dotnet_naming_rule.local_var_and_method_args.style = camel_case
+
+dotnet_naming_rule.private_member_variables.severity = error
+dotnet_naming_rule.private_member_variables.symbols = private_member_variables
+dotnet_naming_rule.private_member_variables.style = pascal_case_with_underscore_prefix

```

Can you change this to apply `_camelCase` instead of `_PascalCase`? I'm not 100% sure that this is the right syntax to do this but it's a starting point:

```suggestion
dotnet_naming_rule.private_member_variables.style = camel_case_with_underscore_prefix
```

---

In this case we don't follow IDesign standard?

Just for reference:

IDesign C# Coding Standard 3.1 on section 1.1.4:

> 4) Prefix private member variables with m_or _. Use Pascal casing for the
rest of a member variable name following the m_ :
class SomeClass
{
int m_Number;
}

Not sure if it's possible to enforce the `m` prefix

---

Yeah, I think in this case we should start with something more conventional -- `int _number` in this example.

There's actually a larger issue as well: there appear to be a couple of areas where the IDesign _.editorconfig_ itself deviates from the IDesign C# Coding Standard. For example, in suggesting `MyType foo = new(...)` syntax which should _never_ be allowed per the standard:
https://github.com/CareTogether/CareTogetherCMS/pull/804/files#diff-2c0a008aa153fd0d1864fe07006720d7f1f6ab3a3cbbe1fd1644f7de7209c5feR131

I've been thinking that it might be wise to look at using something more common/conventional, i.e. [the .NET Runtime C# coding style](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) which has an _.editorconfig_ ([here's the actual current file](https://github.com/dotnet/runtime/blob/main/.editorconfig)) and which is linked to from https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions. The issue with the `= new(...)` syntax changes in particular is troublesome. Let's plan to review this on a call sometime and figure out the best path forward. I know a lot of work has already gone into this PR, and I don't want it to stay out here for any longer than necessary, but I'm a bit concerned about losing code clarity.